ocs_ci/deployment/hosted_cluster.py

```diff
@@ -428,6 +428,82 @@ def deploy_dependencies(
                         f"Failed to install Hypershift on the cluster: {e}"
                     )
 
+        # Enable central infrastructure management service for agent
+        if config.DEPLOYMENT.get("hosted_cluster_platform") == "agent":
+            provisioning_obj = OCP(**OCP(kind=constants.PROVISIONING).get()[0])
+            if not provisioning_obj.data["spec"].get("watchAllNamespaces") == "true":
+                provisioning_obj.patch(
+                    resource_name=provisioning_obj.resource_name,
+                    params='{"spec":{"watchAllNamespaces": true }}',
+                    format_type="merge",
+                )
+                assert (
+                    provisioning_obj.get()["spec"].get("watchAllNamespaces") == "true"
+                ), "Cannot proceed with hosted cluster creation using agent."
+
+            if not OCP(kind=constants.AGENT_SERVICE_CONFIG).get(dont_raise=True):
+                create_agent_service_config()
+            if not OCP(kind=constants.INFRA_ENV).get(dont_raise=True):
+                create_host_inventory()
+
+
+def create_agent_service_config():
+    """
+    Create AgentServiceConfig resource
+
+    """
+    template_yaml = os.path.join(
+        constants.TEMPLATE_DIR, "hosted-cluster", "agent_service_config.yaml"
+    )
+    agent_service_config_data = templating.load_yaml(template_yaml)
+    # TODO: Add custom OS image details
+    helpers.create_resource(**agent_service_config_data)
+
+    # Verify new pods that should be created
+    pod_obj = OCP(kind=constants.POD, namespace="multicluster-engine")
+    for pod_label in ["app=assisted-service", "app=assisted-image-service"]:
```

Comment:
hey, you can use a new utility for this to wait for statuses of the pods simultaneously
[wait_for_pods_to_be_in_statuses_concurrently() ](https://github.com/red-hat-storage/ocs-ci/blob/eb9a669e83c832e5d7374f98beef79758284f566/ocs_ci/ocs/resources/pod.py#L3228)

Reply:
Done